src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -2467,6 +2468,18 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
   return true; // Per default match rules are supported.
 }
 
+const bool Matcher::match_rule_supported_masked_vector(int opcode, int vlen, BasicType bt) {
+  if (!match_rule_supported(opcode) || !vector_size_supported(bt, vlen)) {
```

Comment:
Can we not just use existing routine match_rule_supported_vector here?

Reply:
This can be improved in future. But I think it's different with `match_rule_supported_vector`. It's used to check whether the optimized predicate version is supported/implemented for an operation.

Reply:
I guess there might be different masking support status for operations with SVE and AVX-512. So the check that depends on the architecture should be implemented in the backend.